Competition2018/src/main/java/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,107 @@
+    private final PIDManager headingDrivePid;
+    private final PoseSubsystem poseSubsystem;
+    private final DriveSubsystem drive;
+    private final HeadingModule heading;
+
+    private double deltaDistance;
+    private double targetDistance;
+    private double startingPosition;
+    public final double defaultPValue = 1 / 80d;
+
+    private DoubleProperty deltaDistanceProp;
+    private ContiguousHeading targetHeading;
+
+    @Inject
+    public DriveForDistanceCommand(DriveSubsystem driveSubsystem, XPropertyManager propManager,
+            RobotAssertionManager assertionManager, PIDFactory pidFactory, PoseSubsystem pose, CommonLibFactory clf) {
+
+        this.drive = driveSubsystem;
+        this.poseSubsystem = pose;
+        this.requires(driveSubsystem);
+        this.travelManager = pidFactory.createPIDManager("Drive to position", 0.1, 0, 0, 0, 0.5, -0.5, 3, 1, 0.5);
+        headingDrivePid = pidFactory.createPIDManager("Heading module", defaultPValue, 0, 0);
+        targetHeading = new ContiguousHeading();
+        this.heading = clf.createHeadingModule(headingDrivePid);
+
+    }
+
+    /**
+     * Sets the target distance
+     * 
+     * @param deltaDistance
+     *            the target in inches
+     */
+    public void setDeltaDistance(double deltaDistance) {
+        this.deltaDistance = deltaDistance;
+    }
+
+    /**
+     * Sets the target distance with a DoubleProperty
+     * 
+     * @param deltaDistanceProp
+     *            the DoubleProperty for the target
+     */
+    public void setDeltaDistance(DoubleProperty deltaDistanceProp) {
+        this.deltaDistanceProp = deltaDistanceProp;
+    }
+
+    @Override
+    public void initialize() {
+        startingPosition = getYDistance();
+
+        targetHeading = poseSubsystem.getCurrentHeading();
+
+        if (deltaDistanceProp != null) {
+            this.targetDistance = getYDistance() + deltaDistanceProp.get();
+        } else {
+            this.targetDistance = getYDistance() + deltaDistance;
+        }
+        log.info("Initializing  with distance " + targetDistance + " inches");
+    }
+
+    @Override
+    public void execute() {
+        double power = travelManager.calculate(targetDistance, getYDistance());
+        double headingPower = heading.calculateHeadingPower(targetHeading.getValue());
+
+        double leftPower = power - headingPower;
+        double rightPower = power + headingPower;
+
+        drive.drive(leftPower, rightPower);
+    }
+
+    private double getYDistance() {
+        return poseSubsystem.getRobotOrientedTotalDistanceTraveled().y;
+    }
+
+    @Override
+    public boolean isFinished() {
+        return travelManager.isOnTarget();
+    }
+
+    @Override
+    public void end() {
+        log.info("Ending, PreviousPosition was " + startingPosition + ", Targeted Delta in distance is " + deltaDistance
+                + " Distance traveled is " + (deltaDistance - startingPosition));
```

Comment:
This won't be distance traveled; rather, it'll print... the absolute position you would have been at if you had driven backwards instead of forwards. @JohnGilb

Reply:
@WasabiFan Created an Asana item to track this. If you see stuff like this just put a note in the Programming Kanban - the team is very responsive to bite-size work items in Asana.